src/common/FileSystem.cpp

```diff
@@ -1146,7 +1146,8 @@ static void InternalLoadPak(const PakInfo& pak, Util::optional<uint32_t> expecte
 			if (err)
 				return;
 			depsData.resize(length);
-			zipFile.ReadFile(&depsData[0], length, err);
+			auto read = zipFile.ReadFile(&depsData[0], length, err);
+			depsData.resize(read);
```

Comment:
Hard to say, it depends on how low level we see these routines. If we hard crash lf we make that choice for every caller that nobody can recover from failing to load a pak. And we don't know why we failed. What reason can a zip file read fail on for example, out of memory? Do we won't to die because we can't load a large pak file. Hard failure doesn't seem unreasonable to me in this situation though. I think we should be consistent though, see the earlier error checks. We could perhaps hard fail in those situations too. We should generally try to leave strings with lengths that match their valid data though. That makes debugging easier and reduces risk introducing totally random data into places and if it's just to look at.